polaris-core/src/main/java/org/apache/polaris/core/persistence/LocalPolarisMetaStoreManagerFactory.java

```diff
@@ -181,6 +196,19 @@ private PrincipalSecretsResult bootstrapServiceAndCreatePolarisPrincipalForRealm
       throw new IllegalArgumentException(overrideMessage);
     }
 
+    // TODO rebase onto #422, call a method like PrincipalSecretsGenerator.hasEnvironmentVariables
```

Comment:
idea: maybe pass a flag down to `PrincipalSecretsGenerator` to _not_ use random secrets if `printCredentials` is `false`? Then the `PrincipalSecretsGenerator` can simply throw if the specific realm/user combination is missing env. vars. WDYT?